eng-wasm/src/crypto_wasm.rs

```diff
@@ -15,10 +16,42 @@ pub fn generate_key() -> SymmetricKey {
     key.0
 }
 
-pub fn encrypt(message: &[u8], key: &SymmetricKey, payload: &mut [u8]) {
-    unsafe { external::encrypt(message.as_ptr(), message.len() as u32, key.as_ptr(), payload.as_ptr()) };
+pub fn encrypt(message: &[u8], key: &SymmetricKey) -> Vec<u8> {
+    // TODO: Is this really needed? Dynamically sized buffers don't seem to work.
+    // TODO: Is it possible to estimate the encrypted payload size based on the plaintext message?
+    let length = PAYLOAD_SIZE;
+    let mut payload = Vec::with_capacity(length);
```

Comment:
@moriaab I tried everything that I could to use a dynamic buffer but it only seems to work with a static buffer allocated with zeros. Please confirm that dynamic buffers are not supported. If true, I'll see if I can optimize by estimating the size of the encrypted data instead of using a constant.